### PR TITLE
forforeach/normalize account identifier row parsing

### PR DESCRIPTION
**Pull request by forforeach**
## Intro

Following the BAI2 specifications, the account identifier and summary status sections are compound of two types of components: a common component (customer account number and currency) and summary components (type code, amount, item count, and funds type). Summary components are repeated `n` times. 

## Problem

Some banks emit the last comma at the end of the line and some of them don't. The current parsing mechanism adds a comma by the end of the line in any case. This creates discrepancies and some files are failed to parse because of the redundant comma.

## Proposed solution

When parsing the row into a record, we should normalize the expected amount of commas. We need to be aware of the record type we are working with. We already have the record type, so it is easy. To make the code easier to maintain, I introduced the parsing factory that will map the record type to the appropriate parser. 

Currently, I only implemented the specific parser for the account identifier record (03), but it allows the implementation of normalization for other records easily as needed.

### CODE REVIEW

**Comment by forforeach** [comment]
Hey, @ushkarev, how is it going? My name is Dima, and I work at [Nilus](https://www.nilus.com/). We are using this library, so thank you very much for that. We now have an issue with a couple of banks that we receive data from, and this PR attempts to fix it. 

Is there any chance you could have a look at this?

**Comment by ushkarev** [comment]
Thanks v much for this contribution @forforeach! I’m by no means an expert in bai2 file formats so can’t actually reason about whether this change won’t break something, but am happy to adopt it on the basis that existing tests pass.

I’m a little wary of the addition of typing annotations: they’re not present on any of the other functions/methods. So I would rather they were used everywhere and consistently or not at all. Also, I would bump the minor version rather than the patch level as this is probably a relatively significant change. To that end, I’m going to make a separate branch and PR cherry-picking what you made with a slight tweak – hope you don’t mind :)

File: bai2/helpers.py [hunk 1]
```diff
@@ -17,6 +43,22 @@ def _build_record(rows):
...
 
+RecordBuilderFactory = {
+    RecordCode.file_header: _build_generic_record,
+    RecordCode.group_header: _build_generic_record,
+    RecordCode.account_identifier: _build_account_identifier_record,
+    RecordCode.transaction_detail: _build_generic_record,
+    RecordCode.account_trailer: _build_generic_record,
+    RecordCode.group_trailer: _build_generic_record,
+    RecordCode.file_trailer: _build_generic_record,
+}
```
Comment (ushkarev) [inline bai2/helpers.py:54]: @forforeach Does `RecordCode.continuation` need to be listed here too??

Comment (forforeach) [reply bai2/helpers.py:54]: I think that `RecordCode.continuation` isn't parsed by itself, but only together with one of the other types. That is the reason we have several rows when parsing (there is one the record codes and its continuation rows)

**Comment by forforeach** [comment]
> Thanks v much for this contribution @forforeach! I’m by no means an expert in bai2 file formats so can’t actually reason about whether this change won’t break something, but am happy to adopt it on the basis that existing tests pass.
> 
> I’m a little wary of the addition of typing annotations: they’re not present on any of the other functions/methods. So I would rather they were used everywhere and consistently or not at all. Also, I would bump the minor version rather than the patch level as this is probably a relatively significant change. To that end, I’m going to make a separate branch and PR cherry-picking what you made with a slight tweak – hope you don’t mind :)

Re: the change, I tried to minimize the exposure, thus implemented it only for the specific type that I'm pretty confident about and relyed on the existing tests together with the failing example that also passes now.

Re: typing anotations - that totally makes sense, would you like me to remove them?

Re: version - agreed

**Comment by ushkarev** [comment]
Replacing with #37 which is able to run tests (the actions are set up to not run on forks)

**Pull request closed without merging**